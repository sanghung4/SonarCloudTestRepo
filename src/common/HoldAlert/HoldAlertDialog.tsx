import {
  Box,
  Dialog,
  DialogContent,
  DialogTitle,
  IconButton,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { CloseIcon } from 'icons';

type Props = {
  open: boolean;
  onClose: () => void;
};

function HoldAlertDialog(props: Props) {
  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  /**
   * Render
   */
  return (
    <Dialog
      open={props.open}
      fullWidth
      maxWidth="sm"
      onClose={props.onClose}
      PaperProps={{
        sx: {
          width: 'calc(100% - 32px)',
          my: 4,
          mx: 2
        }
      }}
    >
      <Box
        component={DialogTitle}
        display="flex"
        justifyContent="space-between"
        alignItems="center"
        color="common.white"
        bgcolor="primary.main"
      >
        <Typography variant="h5">{t('common.warning')}</Typography>
        <IconButton
          color="inherit"
          size="small"
          onClick={props.onClose}
          data-testid="hold-alert-dialog-close"
        >
          <CloseIcon />
        </IconButton>
      </Box>
      <DialogContent>
        <Typography
          variant="body1"
          px={isSmallScreen ? 0 : 3}
          py={isSmallScreen ? 1 : 2}
          color="text.secondary"
          whiteSpace={isSmallScreen ? 'initial' : 'pre'}
        >
          {t('common.accountHoldOrdering')}
        </Typography>
      </DialogContent>
    </Dialog>
  );
}

export default HoldAlertDialog;
