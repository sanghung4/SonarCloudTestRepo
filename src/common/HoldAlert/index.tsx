import React, { useContext, useMemo, useState } from 'react';

import {
  Alert,
  Box,
  Collapse,
  IconButton,
  lighten,
  useScreenSize
} from '@dialexa/reece-component-library';

import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';

import { AuthContext } from 'AuthProvider';
import { WarningIcon, CloseIcon } from 'icons';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';

function HoldAlert() {
  /**
   * Custom Hooks
   */
  const location = useLocation();
  const { t } = useTranslation();
  const { isSmallScreen } = useScreenSize();

  /**
   * State
   */
  const [open, setOpen] = useState(true);

  /**
   * Context
   */
  const { authState } = useContext(AuthContext);
  const { selectedAccounts } = useSelectedAccountsContext();

  /**
   * Memo
   */
  const showBanner = useMemo(() => {
    return (
      location.pathname !== '/select-accounts' &&
      (selectedAccounts.billToErpAccount?.creditHold ||
        selectedAccounts.shipToErpAccount?.creditHold) &&
      authState?.isAuthenticated
    );
  }, [
    authState?.isAuthenticated,
    location.pathname,
    selectedAccounts.billToErpAccount?.creditHold,
    selectedAccounts.shipToErpAccount?.creditHold
  ]);

  /**
   * Data
   */
  if (!showBanner) {
    return null;
  }

  return (
    <Box
      zIndex={2}
      top={0}
      position="relative"
      left={isSmallScreen ? undefined : 0}
      right={isSmallScreen ? undefined : 0}
    >
      <Box
        justifyContent="center"
        position={isSmallScreen ? undefined : 'sticky'}
        top={isSmallScreen ? undefined : 0}
        left={isSmallScreen ? undefined : 0}
        right={isSmallScreen ? undefined : 0}
      >
        <Collapse in={open}>
          <Alert
            icon={<WarningIcon />}
            sx={(theme) => ({
              color: 'text.secondary',
              bgcolor: `${lighten(theme.palette.secondary.light, 0.88)}`,
              ...theme.typography.body2,
              alignItems: 'center',
              '& .MuiAlert-icon': {
                color: 'secondary.main'
              },
              border: 1,
              borderColor: 'secondary.main'
            })}
            data-testid="hold-alert"
            action={
              <IconButton
                color="inherit"
                size="small"
                data-testid="close-hold-alert"
                onClick={() => {
                  setOpen(false);
                }}
              >
                <CloseIcon />
              </IconButton>
            }
          >
            {t('common.accountHold')}
          </Alert>
        </Collapse>
      </Box>
    </Box>
  );
}

export default HoldAlert;
