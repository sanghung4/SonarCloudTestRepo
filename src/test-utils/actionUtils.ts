import { fireEvent, findByTestId, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export const fillFormTextInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const fillMaskedFormTextInput = async (
  testId: string,
  value: string
) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await userEvent.type(textInput, value);
  });
};

export const clickButton = async (testId: string) => {
  await act(async () => {
    const button = await findByTestId(document.body, testId);
    await fireEvent.click(button);
  });
};

export const clickRadio = async (testId: string, optionIndex: number) => {
  await act(async () => {
    const radio = await findByTestId(
      document.body,
      `${testId}-radio-${optionIndex}`
    );
    await userEvent.click(radio);
  });
};

export const fillFormSelectInput = async (testId: string, value: string) => {
  await act(async () => {
    const selectInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(selectInput, { target: { value } });
    await fireEvent.keyDown(window, { key: 'Enter' });
  });
};

export const focusInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.focus(input);
  });
};

export const changeInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, testId);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const blurInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.blur(input);
  });
};

export const keyboardDown = async (
  key: string,
  element?: Element | Node | Document | Window
) => {
  await act(async () => {
    await fireEvent.keyDown(element ?? window, { key });
  });
};
