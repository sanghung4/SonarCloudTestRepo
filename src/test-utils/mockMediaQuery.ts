import mediaQuery from 'css-mediaquery';

function createMatchMedia(width: number) {
  return (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: () => jest.fn(),
    removeListener: () => jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  });
}

export const breakpoints = {
  mobile: 414,
  desktop: 1200
};

export const setBreakpoint = (breakpoint: 'mobile' | 'desktop' = 'desktop') => {
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: breakpoints[breakpoint]
  });
  window.dispatchEvent(new Event('resize'));

  window.matchMedia = createMatchMedia(window.innerWidth);
};
