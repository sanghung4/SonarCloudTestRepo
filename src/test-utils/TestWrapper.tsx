import React, { JSXElementConstructor, ReactElement, ReactNode } from 'react';

import { ThemeProvider } from '@dialexa/reece-component-library';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { OktaAuth } from '@okta/okta-auth-js';
import { render as jestRender, RenderResult } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';

import { UserContextType, AuthContext } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';
import i18n from 'test-utils/mockI18N';
import {
  SelectedAccountsContext,
  SelectedAccountsContextType
} from 'providers/SelectedAccountsProvider';
import { CartContext, CartContextType } from 'providers/CartProvider';
import {
  CheckoutContext,
  CheckoutContextType,
  initialCheckoutContextValues
} from 'Checkout/CheckoutProvider';

/**
 * NOTE:
 * To setup mocks follow the examples here:
 * https://www.apollographql.com/docs/react/development-testing/testing/
 *
 * To get a successful response in your test (not just the loading state)
 * you can do the following:
 *
 * it('Should render', async () => {
 *   const { container } = renderMockRouter(<MyComponent />, mocks);
 *
 *   await waitFor(() => new Promise((res) => setTimeout(res, 0)));
 *
 *   expect(container).toMatchSnapshot();
 * });
 *
 */

/******************************/
/* Types                      */
/******************************/
type Props = {
  children?: ReactNode;
  authConfig?: Partial<UserContextType>;
  selectedAccountsConfig?: Partial<SelectedAccountsContextType>;
  cartConfig?: Partial<CartContextType>;
  checkoutConfig?: Partial<CheckoutContextType>;
  mocks?: ReadonlyArray<MockedResponse>;
};

type WrapperProps<TKey extends keyof Props> = {
  value: Props[TKey];
  children: React.ReactNode;
};

/******************************/
/* Defaults                   */
/******************************/
const defaultAuth: UserContextType = {
  authState: {
    isAuthenticated: true
  },
  isLoggingOut: false,
  oktaAuth: new OktaAuth({ issuer: 'http://test.com' }),
  profile: {
    userId: 'testuser',
    permissions: Object.values(Permission),
    isEmployee: false,
    isVerified: false
  }
};

const defaultSelectedAccounts: SelectedAccountsContextType = {
  clearAccounts: () => {},
  updateAccounts: () => {},
  updateShippingBranchId: () => {},
  loading: false,
  selectedAccounts: {
    billTo: { erpAccountId: '' },
    shipTo: { erpAccountId: '' },
    shippingBranchId: ''
  },
  isMincron: false,
  isEclipse: false
};

const defaultCartContextValue: CartContextType = {
  addItemToCart: () => {},
  addItemsToCart: () => {},
  addAllListItemsToCart: () => {},
  cart: undefined,
  cartLoading: false,
  checkingOutWithQuote: false,
  contractBranch: undefined,
  clearContract: () => {},
  clearQuote: () => {},
  deleteCartItems: () => new Promise(() => {}),
  disableAddToCart: false,
  deleteItem: () => {},
  getUserCart: () => new Promise(() => {}),
  itemCount: 0,
  itemLoading: undefined,
  lineNotes: {},
  releaseContractToCart: () => {},
  setContract: () => {},
  setLineNotes: () => {},
  setQuoteId: () => {},
  setQuoteData: () => {},
  setQuoteShipToId: () => {},
  setPreviousCart: () => {},
  setSelectedBranch: () => {},
  updateCart: undefined,
  updateDelivery: undefined,
  updateItemQuantity: () => {},
  updateWillCall: undefined,
  updateWillCallBranch: undefined,
  refreshCart: undefined,
  isWillCall: false,
  deleteCreditCardFromCart: () => new Promise(() => {})
};

/******************************/
/* Wrappers                   */
/******************************/
function SelectedAccountsProviderWrapper({
  value,
  children
}: WrapperProps<'selectedAccountsConfig'>) {
  if (value) {
    return (
      <SelectedAccountsContext.Provider
        value={{ ...defaultSelectedAccounts, ...value }}
      >
        {children}
      </SelectedAccountsContext.Provider>
    );
  }

  return <>{children}</>;
}

function CartProviderWrapper({ value, children }: WrapperProps<'cartConfig'>) {
  if (value) {
    return (
      <CartContext.Provider value={{ ...defaultCartContextValue, ...value }}>
        {children}
      </CartContext.Provider>
    );
  }

  return <>{children}</>;
}

function CheckoutProviderWrapper({
  value,
  children
}: WrapperProps<'checkoutConfig'>) {
  if (value) {
    return (
      <CheckoutContext.Provider
        value={{ ...initialCheckoutContextValues, ...value }}
      >
        {children}
      </CheckoutContext.Provider>
    );
  }

  return <>{children}</>;
}

function TestWrapper(props: Props) {
  return (
    <ThemeProvider>
      <I18nextProvider i18n={i18n}>
        <AuthContext.Provider value={{ ...defaultAuth, ...props.authConfig }}>
          <MockedProvider mocks={props.mocks ?? []}>
            <SelectedAccountsProviderWrapper
              value={props.selectedAccountsConfig}
            >
              <CartProviderWrapper value={props.cartConfig}>
                <CheckoutProviderWrapper value={props.checkoutConfig}>
                  {props.children}
                </CheckoutProviderWrapper>
              </CartProviderWrapper>
            </SelectedAccountsProviderWrapper>
          </MockedProvider>
        </AuthContext.Provider>
      </I18nextProvider>
    </ThemeProvider>
  );
}

function jestRenderFn(wrapped: JSX.Element) {
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined
  });
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRenderFn(wrapped);
}
export type TestRender = RenderResult<
  typeof import('@testing-library/dom/types/queries'),
  Element | DocumentFragment
>;

export function renderWithComponent(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return {
    render: jestRenderFn(wrapped),
    component: wrapped
  };
}

export default TestWrapper;
