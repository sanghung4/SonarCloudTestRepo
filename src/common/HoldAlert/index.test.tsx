import HoldAlert from 'common/HoldAlert';
import HoldAlertDialog from 'common/HoldAlert/HoldAlertDialog';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';
import { fireEvent, screen, waitFor } from '@testing-library/react';

describe('common - HoldAlert', () => {
  it('Should match snapshot when account is not on hold', () => {
    const { container } = render(<HoldAlert />, {
      selectedAccountsConfig: {
        selectedAccounts: {
          billToErpAccount: { creditHold: false }
        }
      }
    });
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot when shipTo account is not on hold', () => {
    const { container } = render(<HoldAlert />, {
      selectedAccountsConfig: {
        selectedAccounts: {
          shipToErpAccount: { creditHold: false }
        }
      }
    });
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot on desktop', () => {
    setBreakpoint('desktop');
    const { container } = render(<HoldAlert />, {
      selectedAccountsConfig: {
        selectedAccounts: {
          billToErpAccount: { creditHold: true }
        }
      }
    });
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot on mobile', () => {
    setBreakpoint('mobile');
    const { container } = render(<HoldAlert />, {
      selectedAccountsConfig: {
        selectedAccounts: {
          billToErpAccount: { creditHold: true }
        }
      }
    });
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot on desktop for shipTo account', () => {
    setBreakpoint('desktop');
    const { container } = render(<HoldAlert />, {
      selectedAccountsConfig: {
        selectedAccounts: {
          shipToErpAccount: { creditHold: true }
        }
      }
    });
    expect(container).toMatchSnapshot();
    expect(container).toBeInTheDocument();
  });

  it('Should match snapshot on mobile for shipTo account', () => {
    setBreakpoint('mobile');
    const { container } = render(<HoldAlert />, {
      selectedAccountsConfig: {
        selectedAccounts: {
          shipToErpAccount: { creditHold: true }
        }
      }
    });
    expect(container).toMatchSnapshot();
  });

  it('Should close Alert on clicking the cross-icon', async () => {
    setBreakpoint('desktop');
    const { container } = render(<HoldAlert />, {
      selectedAccountsConfig: {
        selectedAccounts: {
          shipToErpAccount: { creditHold: true }
        }
      }
    });
    expect(container).toBeInTheDocument();

    const closeIcon = screen.getByTestId('close-hold-alert');
    const holdAlert = screen.getByTestId('hold-alert');

    fireEvent.click(closeIcon);

    await waitFor(() => {
      expect(holdAlert).not.toBeVisible();
    });
    expect(container).toBeInTheDocument();
  });
});

describe('common - HoldAlert Dialog', () => {
  it('Should match snapshot on desktop', () => {
    setBreakpoint('desktop');
    const wrapper = render(<HoldAlertDialog open onClose={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match snapshot on mobile', () => {
    setBreakpoint('mobile');
    const wrapper = render(<HoldAlertDialog open onClose={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
