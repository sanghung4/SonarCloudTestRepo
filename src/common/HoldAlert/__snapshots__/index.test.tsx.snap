// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common - HoldAlert Dialog Should match snapshot on desktop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1200px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-15p4qre-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="mui-1"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-c2jo8y-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root MuiBox-root css-c3i9us-MuiTypography-root-MuiDialogTitle-root"
            id="mui-1"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-16xaac9-MuiTypography-root"
            >
              Warning
            </h5>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ff60md-MuiButtonBase-root-MuiIconButton-root"
              data-testid="hold-alert-dialog-close"
              tabindex="0"
              type="button"
            >
              <svg>
                delete.svg
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </h2>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-7tupzl-MuiTypography-root"
            >
              Ordering on this account is currently disabled.
Please contact the credit team.
            </p>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`common - HoldAlert Dialog Should match snapshot on mobile 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 414px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-15p4qre-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="mui-2"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-c2jo8y-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root MuiBox-root css-c3i9us-MuiTypography-root-MuiDialogTitle-root"
            id="mui-2"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-16xaac9-MuiTypography-root"
            >
              Warning
            </h5>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ff60md-MuiButtonBase-root-MuiIconButton-root"
              data-testid="hold-alert-dialog-close"
              tabindex="0"
              type="button"
            >
              <svg>
                delete.svg
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </h2>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1rb87yv-MuiTypography-root"
            >
              Ordering on this account is currently disabled.
Please contact the credit team.
            </p>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`common - HoldAlert Should match snapshot on desktop 1`] = `
<div>
  <div
    class="MuiBox-root css-hs5cmw"
  >
    <div
      class="MuiBox-root css-fyhf7q"
    >
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-b7xqgi-MuiPaper-root-MuiAlert-root"
              data-testid="hold-alert"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
              >
                <svg>
                  warning.svg
                </svg>
              </div>
              <div
                class="MuiAlert-message css-acap47-MuiAlert-message"
              >
                There may be an issue with your account. Please contact the Credit Department.
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ff60md-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="close-hold-alert"
                  tabindex="0"
                  type="button"
                >
                  <svg>
                    delete.svg
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - HoldAlert Should match snapshot on desktop for shipTo account 1`] = `
<div>
  <div
    class="MuiBox-root css-hs5cmw"
  >
    <div
      class="MuiBox-root css-fyhf7q"
    >
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-b7xqgi-MuiPaper-root-MuiAlert-root"
              data-testid="hold-alert"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
              >
                <svg>
                  warning.svg
                </svg>
              </div>
              <div
                class="MuiAlert-message css-acap47-MuiAlert-message"
              >
                There may be an issue with your account. Please contact the Credit Department.
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ff60md-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="close-hold-alert"
                  tabindex="0"
                  type="button"
                >
                  <svg>
                    delete.svg
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - HoldAlert Should match snapshot on mobile 1`] = `
<div>
  <div
    class="MuiBox-root css-dh91i8"
  >
    <div
      class="MuiBox-root css-1tyndxa"
    >
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-b7xqgi-MuiPaper-root-MuiAlert-root"
              data-testid="hold-alert"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
              >
                <svg>
                  warning.svg
                </svg>
              </div>
              <div
                class="MuiAlert-message css-acap47-MuiAlert-message"
              >
                There may be an issue with your account. Please contact the Credit Department.
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ff60md-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="close-hold-alert"
                  tabindex="0"
                  type="button"
                >
                  <svg>
                    delete.svg
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - HoldAlert Should match snapshot on mobile for shipTo account 1`] = `
<div>
  <div
    class="MuiBox-root css-dh91i8"
  >
    <div
      class="MuiBox-root css-1tyndxa"
    >
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-b7xqgi-MuiPaper-root-MuiAlert-root"
              data-testid="hold-alert"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
              >
                <svg>
                  warning.svg
                </svg>
              </div>
              <div
                class="MuiAlert-message css-acap47-MuiAlert-message"
              >
                There may be an issue with your account. Please contact the Credit Department.
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ff60md-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="close-hold-alert"
                  tabindex="0"
                  type="button"
                >
                  <svg>
                    delete.svg
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`common - HoldAlert Should match snapshot when account is not on hold 1`] = `<div />`;

exports[`common - HoldAlert Should match snapshot when shipTo account is not on hold 1`] = `<div />`;
